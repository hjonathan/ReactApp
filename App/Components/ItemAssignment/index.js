import React, {Component} from "react";
import {View, Text} from "react-native";
import I18n from "../../I18n";
import Styles from "./styles";

/**
 * Class ItemAssignment
 * Example:
 * import ItemList from '../ItemList';
 * render() {
 *      return(
 *          <ItemAssignment data={object}/>
 *      ),
 * }
 */
class ItemAssignment extends Component {
    /**
     * Constructor ItemAssignment
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            data: (props && props.data) || [],
            users: ""
        };
    }

    /**
     * Prepares the user  info, concats the array users
     */
    componentWillMount () {
        const that = this,
            // prepare the users info "A, B, C"
            users = Array.prototype.map.call(
                this.state.data.users,
                (item) => {
                    const {taskId} = that.state.data;
                    that.props.addUserId({[taskId]: item.userId});
                    return item.userFullName;
                }
            ).join(", ");
        this.setState({users});
    }

    /**
     * Render the user
     */
    render () {
        return (
            <View>
                <View style={Styles.firstSection}>
                    <View>
                        <Text style={Styles.text}>
                            {I18n.t("task_name_label_user_route_screen")} {this.state.data.taskName || I18n.t("title_user_route_screen_end_of_process")}
                        </Text>
                        <Text style={Styles.text}>
                            {I18n.t("next_user_label_users_route_screen")} {this.state.users}
                        </Text>
                    </View>
                </View>
                <View style={Styles.dividerLine} />
            </View>
        );
    }
}

export default ItemAssignment;
