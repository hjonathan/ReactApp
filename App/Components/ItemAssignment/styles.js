import {StyleSheet} from "react-native";
import * as Res from "../../Assets/resources";

export default StyleSheet.create({
    text: {
        color: "black",
        textAlign: "center",
        fontSize: 14,
        marginHorizontal: "10%"
    },
    dividerLine: {
        flex: 1,
        backgroundColor: Res.colors.gray,
        marginHorizontal: 8,
        height: 1
    }
});
