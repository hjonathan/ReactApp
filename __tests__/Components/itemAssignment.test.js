import 'react-native';
import React from 'react';
import ItemAssignment from '../../App/Components/ItemAssignment';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Test of item assignment for list', () => {
    let container;
    const data = {
            users: [
                {
                    userId: "00000001",
                    userFullName: "Test 0001"
                }
            ],
            taskId: "10000000",
        },
        addUserId = jest.fn();

    beforeEach(() => {
        container = shallow(
            <ItemAssignment
                data={data}
                addUserId={addUserId}
            />
        );
    });

    it('render Item correctly', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('Render the connected(SMART) component', () => {
        expect(container.length).toEqual(1)
    });

    it('Check Prop matches with data', () => {
        expect(container.instance().state.data).toEqual(data);
    });

    it('addUserId function', () => {
        expect(container.instance().props.addUserId()).toMatchSnapshot();
    });
});
