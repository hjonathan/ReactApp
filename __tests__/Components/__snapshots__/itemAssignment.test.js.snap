// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of item assignment for list addUserId function 1`] = `undefined`;

exports[`Test of item assignment for list render Item correctly 1`] = `
<Component>
  <Component>
    <Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 14,
            "marginHorizontal": "10%",
            "textAlign": "center",
          }
        }
      >
        task_name_label_user_route_screen
         
        title_user_route_screen_end_of_process
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 14,
            "marginHorizontal": "10%",
            "textAlign": "center",
          }
        }
      >
        next_user_label_users_route_screen
         
        Test 0001
      </Text>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "backgroundColor": "#BABABA",
        "flex": 1,
        "height": 1,
        "marginHorizontal": 8,
      }
    }
  />
</Component>
`;
